src/javax/bluetooth/UUID.java

```diff
@@ -0,0 +1,83 @@
+/*
+	This file is part of FreeJ2ME.
+
+	FreeJ2ME is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	FreeJ2ME is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with FreeJ2ME.  If not, see http://www.gnu.org/licenses/
+*/
+package javax.bluetooth;
+
+public class UUID
+{
+	
+	private final long[] baseUUID = { 0x0000_0000L, 0x0000_1000L, 0x8000_0080L, 0x5F9B_34FBL };
+
+	private final long[] UUIDval  = { 0x0000_0000L, 0x0000_0000L, 0x0000_0000L, 0x0000_0000L };
+
+
+	public UUID(long uuidValue) throws IllegalArgumentException
+	{
+		if (uuidValue < 0L || uuidValue > 0xffffffffL) { throw new IllegalArgumentException("Invalid UUID Value, must be between 0 an (2^32)-1"); }
+		else
+		{
+			UUIDval[0] = uuidValue;
+			UUIDval[1] = baseUUID[1];
+			UUIDval[2] = baseUUID[2];
+			UUIDval[3] = baseUUID[3];
+		}
+	}
+
+	public UUID(String uuidValue, boolean shortUUID) throws IllegalArgumentException, NullPointerException, NumberFormatException
+	{ 	
+		int length = uuidValue.length();
+
+		if(uuidValue == null || length==0 || length>32 || (shortUUID && length>8) )
```

Comment:
linLe 41 should throw a `NullPopinterException` if `uuidValue` is null.  There isn't a reason to check here.